modules/ocf_kvm/facts.d/vms

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+import json
+import subprocess
+
+
+def run_command(command_args):
+    output = subprocess.check_output(command_args).decode('utf-8').split('\n')
+
+    # Return the command output split by line with any empty lines removed
+    return list(filter(None, output))
+
+
+vms_all = run_command(('/usr/bin/virsh', 'list', '--name', '--all'))
+vms_inactive = run_command(('/usr/bin/virsh', 'list', '--name', '--inactive'))
+
+print(json.dumps(
+    {
+        'vms': vms_all,
+        'vms_off': vms_inactive,
+        'vms_on': list(set(vms_all) - set(vms_inactive)),
+    },
```

Comment:
it might be worth sorting these lists when printing out. it won't matter for programs, but could be useful for staff who run "facter" or whatever

Reply:
I was thinking of doing that too, but I thought this order might be useful too, because it indicates which order the VMs were started in on the host. I also thought was also that it would be pretty easy to sort if needed, so leaving it with this order gives a bit more information.